### PR TITLE
add offset Y

### PR DESCRIPTION
**Pull request by saidReclip**
### 🎯 Goal

We need to observe offset Y of ChannelList

### CODE REVIEW

**Comment by saidReclip** [comment]
Hello @nuno-vieira! 
Could we get a review on this?

File: Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelList.swift [hunk 1]
```diff
@@ -56,13 +58,24 @@ public struct ChannelList<Factory: ViewFactory>: View {
...
         _swipedChannelId = swipedChannelId
+        _offsetY = offsetY
     }
     
     public var body: some View {
         Group {
             if scrollable {
                 ScrollView {
-                    channelsVStack
+                    ZStack {
```
Comment (martinmitrevski) [inline Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelList.swift:68]: Thanks for the PR. Not sure if this should be enabled by default for all users. We should probably have a config with an opt-in. Will think about this and do a follow-up.

**Comment by nuno-vieira** [comment]
> Hello @nuno-vieira! Could we get a review on this?

@saidReclip, @martinmitrevski will do it when he has availability

**Comment by martinmitrevski** [comment]
Hey @saidReclip,

Thanks again for creating the PR. Unfortunately, this introduces noticeable performance degradation in the scrolling experience (especially with many channels), therefore we can't merge it in. 

Instead, I've made the `ChannelsLazyVStack` public, so you can do this in your app (in case you don't have these performance issues).

Best,
Martin

**Pull request closed without merging**